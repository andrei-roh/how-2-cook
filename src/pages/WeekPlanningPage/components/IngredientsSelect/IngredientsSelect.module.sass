@use 'src/style/colors'
@use 'src/style/breakpoints'
@import 'src/style/typography'

.selectPlaceholder
    @extend .T1
    color: colors.$gray

div .submitButton
    @extend .H6
    width: 147px
    height: 48px
    box-sizing: border-box
    color: colors.$white
    background-color: colors.$blue

    &:hover
        background-color: colors.$sky

    &:disabled,
    &[disabled]
        cursor: auto 
        background-color: colors.$gray