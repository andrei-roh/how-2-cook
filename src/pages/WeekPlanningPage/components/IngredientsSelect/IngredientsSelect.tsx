import CircularProgress from '@mui/material/CircularProgress';
import Button from '@mui/material/Button';
import { Select } from 'src/components';
import { getIngredientsToSelect } from 'src/utils/ingredients/getIngredientsToSelect';
import { IIngredient } from 'src/types';
import css from './IngredientsSelect.module.sass';
import { isArraysEqual } from 'src/utils/isArraysEqual';

interface IngredientsSelectProps {
  recipeIngredients: string[] | [];
  setRecipeIngredients: React.Dispatch<React.SetStateAction<[] | string[]>>;
  allIngredients: IIngredient[];
  handleGetRecipesByIngredient: () => void;
  isPlanning: boolean;
  planningIngredients: string[];
}

export const IngredientsSelect = ({
  recipeIngredients,
  setRecipeIngredients,
  allIngredients,
  handleGetRecipesByIngredient,
  isPlanning,
  planningIngredients,
}: IngredientsSelectProps) => {
  return (
    <>
      <Select
        value={recipeIngredients}
        setChange={setRecipeIngredients}
        options={getIngredientsToSelect(allIngredients)}
        labelText={'Ингредиенты'}
        multiple
        placeholder={'Выбор'}
        selectClassName={!recipeIngredients ? css.selectPlaceholder : undefined}
      />
      <Button
        className={css.submitButton}
        onClick={handleGetRecipesByIngredient}
        disabled={
          isPlanning || isArraysEqual(recipeIngredients, planningIngredients)
        }
      >
        {isPlanning ? (
          <CircularProgress color='inherit' size={32} />
        ) : (
          'Спланировать'
        )}
      </Button>
    </>
  );
};
