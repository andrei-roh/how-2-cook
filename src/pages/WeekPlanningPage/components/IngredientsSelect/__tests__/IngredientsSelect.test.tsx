import { describe, expect, it, vi } from 'vitest';
import { render } from '@testing-library/react';
import { IngredientsSelect } from '../IngredientsSelect';

const recipeIngredientsMock = ['00000000-0000-0000-0000-000000000001'];
const setRecipeIngredientsMock = vi.fn();
const allIngredientsMock = [
  {
    id: '00000000-0000-0000-0000-000000000000',
    name: 'Ingredient 1',
  },
  {
    id: '00000000-0000-0000-0000-000000000001',
    name: 'Ingredient 2',
  },
  {
    id: '00000000-0000-0000-0000-000000000002',
    name: 'Ingredient 3',
  },
  {
    id: '00000000-0000-0000-0000-000000000003',
    name: 'Ingredient 4',
  },
];
const handleGetRecipesByIngredientMock = vi.fn();
const planningIngredientsMock = [
  '00000000-0000-0000-0000-000000000002',
  '00000000-0000-0000-0000-000000000003',
];

describe('Testing: IngredientsSelect', () => {
  it('should render', () => {
    const { container } = render(
      <IngredientsSelect
        recipeIngredients={recipeIngredientsMock}
        setRecipeIngredients={setRecipeIngredientsMock}
        allIngredients={allIngredientsMock}
        handleGetRecipesByIngredient={handleGetRecipesByIngredientMock}
        isPlanning={false}
        planningIngredients={planningIngredientsMock}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
