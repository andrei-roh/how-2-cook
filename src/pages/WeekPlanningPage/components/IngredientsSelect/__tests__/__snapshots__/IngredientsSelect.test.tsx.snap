// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing: IngredientsSelect > should render 1`] = `
<div>
  <label
    class="_label_9fe140"
  >
    <div>
      Ингредиенты
    </div>
    <div
      class="_select_9fe140"
      tabindex="0"
    >
      INGREDIENT 2
      <img
        class="_selectArrow_9fe140"
        src="/src/components/Select/assets/arrow.svg"
      />
    </div>
    <div
      class="_selectMenuWrapper_9fe140"
    />
  </label>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary _submitButton_362ec8 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Спланировать
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
